kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/serializer/ConfigTokenSerializer.kt

```diff
@@ -0,0 +1,49 @@
+package org.axonframework.extensions.kotlin.serialization.serializer
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.Serializer
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.descriptors.element
+import kotlinx.serialization.encoding.CompositeDecoder.Companion.DECODE_DONE
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+import org.axonframework.eventhandling.tokenstore.ConfigToken
+
+@OptIn(ExperimentalSerializationApi::class)
```

Comment:
Would you be able to elaborate a little on the `ExperimentalSerializationApi` you're "opting in" here?

Reply:
The `@Serializer` annotation below is marked as experimental. It might change its implementation slightly between versions of Kotlin Serialization. (https://kotlin.github.io/kotlinx.serialization/kotlinx-serialization-core/kotlinx-serialization-core/kotlinx.serialization/-experimental-serialization-api/index.html)

We might wait to merge this feature until the used serialization API is marked fully stable, although the Axon `extension-kotlin` library  is also marked as experimental itself.

Reply:
Yes exactly. The following is stable:
```
@Serializable
class XXX
```

But the following needs experimental opt in:
```
@Serializer(forClass = QQQ::class)
class XXX
```

In particular doing this for Java classes gives compiler errors, and it might simply not be supported (see https://github.com/Kotlin/kotlinx.serialization/issues/1687)